src/main/java/com/iota/iri/service/tipselection/impl/WalkerAlpha.java

```diff
@@ -0,0 +1,146 @@
+package com.iota.iri.service.tipselection.impl;
+
+import com.iota.iri.controllers.ApproveeViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.model.HashId;
+import com.iota.iri.service.tipselection.TailFinder;
+import com.iota.iri.service.tipselection.WalkValidator;
+import com.iota.iri.service.tipselection.Walker;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.utils.collections.interfaces.UnIterableMap;
+import com.iota.iri.zmq.MessageQ;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of <tt>Walker</tt> that performs a weighted random walk
+ * with <CODE>e^(alpha*Hy)</CODE> as the transition function.
+ *
+ */
+public class WalkerAlpha implements Walker {
+
+    private double alpha;
+    private final Random random;
+
+    private final Tangle tangle;
+    private final MessageQ messageQ;
+    private final Logger log = LoggerFactory.getLogger(Walker.class);
+
+    private final TailFinder tailFinder;
+
+    public WalkerAlpha(double alpha, Random random, Tangle tangle, MessageQ messageQ, TailFinder tailFinder) {
+        this.alpha = alpha;
+        this.random = random;
+        this.tangle = tangle;
+        this.messageQ = messageQ;
+        this.tailFinder = tailFinder;
+    }
+
+    public double getAlpha() {
+        return alpha;
+    }
+
+    public void setAlpha(double alpha) {
+        this.alpha = alpha;
+    }
+
+    @Override
+    public Hash walk(Hash entryPoint, UnIterableMap<HashId, Integer> ratings, WalkValidator walkValidator) throws Exception {
+        if (!walkValidator.isValid(entryPoint)) {
+            throw new RuntimeException("entry point failed consistency check: " + entryPoint.toString());
```

Comment:
@alon-e we need to decide between:
1. suppress this warning
2. Use `IllegalStateException`
3. Create a custom error.

Btw, the error message is incorrect.. The entry point could be invalid because of several reasons. Not just "consistency".

Reply:
In the meanwhile I am throwing `IllegalStateException`